service/service.go

```diff
@@ -102,6 +103,12 @@ func (svc *Service) Start(ctx context.Context, svcErrors chan error) {
 			svcErrors <- fmt.Errorf("failed to start main http server: %w", err)
 		}
 	}()
+
+	// wait for producer to be initialised and consumer to be in consuming state
+	<-svc.producer.Channels().Initialised
+	log.Info(ctx, "kafka producer initialised")
+	<-svc.consumer.Channels().State.Consuming
```

Comment:
I thought this func could exit without requiring the producer to be initialised or the consumer to be consuming, is there any reason why you added this? (maybe it makes sense to add it in the component tests?)

Reply:
So yeah this was the other half of the component test fix - I guessed it was the Kafka consumer not being ready before the component tests started which was the reason the first kafka message was not being picked up. Then the channel was added to guarantee that this is completed. 
There isn't an immediate way to get access to the consumer/producer from the component tests as they're not exported, I wondered why we weren't waiting for them to be ready as standard though as they need to be ready for the service to be of any use?

Reply:
I did some extra testing, and this may cause problems because we block the main thread.
For example, if kafka is not ready, we whould block the main thread and wouldn't be able to run a graceful shutdown. The process would be left as a zombie:

I created this PR aganst your branch to fix it:
https://github.com/ONSdigital/dp-cantabular-metadata-exporter/pull/29

But maybe we should just export the service Producer and Consumer, and copy this block of code to the component tests? (it would make the clode cleaner). If we don't want to export it, maybe we can use getters.